qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,446 @@
+                 │  cr_tone(duration) │┌─┴───┴─┐│  cr_tone(duration) │┌────────┐
+            q_1: ┤1                   ├┤ Rz(π) ├┤1                   ├┤ Rz(-π) ├
+                 └────────────────────┘└───────┘└────────────────────┘└────────┘
+
+        Here two ``cr_tone``s are applied where the latter one is with the
+        control qubit state flipped and with a phase flip of the target qubit frame.
+        This operation is equivalent to applying the ``cr_tone`` with a negative amplitude.
+        The Hamiltonian for this decomposition has no IX and ZI interactions,
+        and also a reduced IY interaction to some extent (not completely eliminated) [1].
+        Note that the CR Hamiltonian tomography experiment cannot detect the ZI term.
+        However, it is sensitive to the IX and IY terms.
+
+    # section: reference
+        .. ref_arxiv:: 1 2007.02925
+
+    """
+
+    def _build_cr_circuit(
+        self,
+        backend: Backend,
+        flat_top_width: circuit.Parameter,
+        sigma: float,
+    ) -> QuantumCircuit:
+        """Echoed cross resonance.
+
+        Args:
+            backend: The target backend.
+            flat_top_width: Total length of flat top part of the pulse in units of dt.
+            sigma: Sigma of Gaussian edges in units of dt.
+
+        Returns:
+            A circuit definition for the cross resonance pulse to measure.
+        """
+        flat_top_width /= 2
+
+        cr_gate = circuit.Gate("cr_tone", num_qubits=2, params=[flat_top_width])
+
+        cr_circuit = QuantumCircuit(2)
+        cr_circuit.append(cr_gate, [0, 1])
+        cr_circuit.x(0)
+        cr_circuit.rz(np.pi, 1)
+        cr_circuit.append(cr_gate, [0, 1])
+        cr_circuit.rz(-np.pi, 1)
+
+        opt = self.experiment_options
+
+        # Compute valid integer duration
+        cr_duration = round_pulse_duration(
+            backend=backend, duration=flat_top_width + 2 * sigma * opt.risefall
+        )
+
+        with pulse.build(backend, default_alignment="left", name="cr") as cross_resonance:
+            # add cross resonance tone
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration=cr_duration,
+                    amp=opt.amp,
+                    sigma=sigma,
+                    width=flat_top_width,
+                ),
+                pulse.control_channels(*self.physical_qubits)[0],
+            )
+            # add cancellation tone
+            if not np.isclose(opt.amp_t, 0.0):
+                pulse.play(
+                    pulse.GaussianSquare(
+                        duration=cr_duration,
+                        amp=opt.amp_t,
+                        sigma=sigma,
+                        width=flat_top_width,
+                    ),
+                    pulse.drive_channel(self.physical_qubits[1]),
+                )
+            else:
+                pulse.delay(cr_duration, pulse.drive_channel(self.physical_qubits[1]))
```

Comment:
This looks identical to lines 209 to 233. Can we have this in a function in the `CrossResonanceHamiltonian` class?

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/385/commits/15e021062f469c502e24324ee072876aa2e1bbd1. Note that subclass cannot define a CR schedule with multiple kinds of CR pulses.